### PR TITLE
Re-organize the Link022 repo.

### PR DESCRIPTION
**Pull request by boleifu**
Change Summary:
1. Updated README files to separate out each components.
2. Added Link022 emulator.
3. Sync to the latest OpenConfig model.
4. Fixed incorrect model name in sample config files.
5. Added gNMI test kit.

### CODE REVIEW

File: testkit/gnmiconfig/worker.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+		var actualJson, expectedJson interface{}
+		if err := json.Unmarshal(actualJsonValue.JsonIetfVal, &actualJson); err != nil {
+			return fmt.Errorf("invalid value %v: %v", string(actualJsonValue.JsonIetfVal), err)
+		}
+		if err := json.Unmarshal(expectedJsonValue.JsonIetfVal, &expectedJson); err != nil {
+			return fmt.Errorf("invalid value %v: %v", string(expectedJsonValue.JsonIetfVal), err)
+		}
+		if !reflect.DeepEqual(actualJson, expectedJson) {
+			log.Errorf("actual=\n%v", string(actualJsonValue.JsonIetfVal))
+			log.Errorf("expected=\n%v", string(expectedJsonValue.JsonIetfVal))
```
Comment (zhiweicen) [inline testkit/gnmiconfig/worker.go]: Why do we need the two lines above? (the information is included in the err msg below).

File: testkit/test_kit.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+var (
+	gnmiTests  arrayFlags
+	targetAddr = flag.String("target_addr", "localhost:10161", "The target address in the format of host:port")
+	targetName = flag.String("target_name", "hostname.com", "The target name use to verify the hostname returned by TLS handshake")
```
Comment (zhiweicen) [inline testkit/test_kit.go]: s/use to/used to/

File: agent/README.md [hunk 3]
```diff
@@ -1,4 +1,72 @@
...
 This directory contains the WiFi management component that runs on the
-Link022 Pi AP.
\ No newline at end of file
+Link022 Pi AP.
+
+## Get Started
+For documentation on installing a full demo system (with both a gNMI client and link022 access point)
+please see the [full demo documentation](../demo/README.md) which explains a fully complete system.
+
+The following instructions will get you a Link022 AP on a Unix-based device.
```
[the remark this answers is not in the document]
Comment (boleifu) [reply agent/README.md]: Done

File: agent/README.md [hunk 4]
```diff
@@ -1,4 +1,72 @@
...
+
+### Prerequisites
+Have a device set up. (Tested with Raspbian)
```
[the remark this answers is not in the document]
Comment (boleifu) [reply agent/README.md]: Done

File: agent/README.md [hunk 5]
```diff
@@ -1,4 +1,72 @@
...
+
+### Download Link022 agent
+```
+export GOPATH=$HOME/go
+go get github.com/google/link022/agent
+```
+
+### Download certificates
+Download sample certificates from [the demo directory](../demo/cert/server/).
+Or you can use your own cert.
```
[the remark this answers is not in the document]
Comment (boleifu) [reply agent/README.md]: Done

File: build_pi.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (boleifu) [reply build_pi.sh]: Done

File: build_pi.sh [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+
+# Build Link022 agent for Raspberry Pi.
```
[the remark this answers is not in the document]
Comment (boleifu) [reply build_pi.sh]: Done

File: emulator/README.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+```
+mininet> nodes
+available nodes are: 
+ctrlr dummy target
+```
+
+There are three nodes in mininet.
+* ctrlr: where gNMI client runs on.
+* target: where Link022 agent runs on.
+* dummy: a dummy host connects to the emulated Link022 AP.
```
[the remark this answers is not in the document]
Comment (boleifu) [reply emulator/README.md]: Done

File: emulator/emulator.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
+"""Emulator for link022. Refer to the README.md file for instructions.
```
[the remark this answers is not in the document]
Comment (boleifu) [reply emulator/emulator.py]: Done

File: testkit/gnmiconfig/worker.go (hunk 1 above)
Comment (boleifu) [reply testkit/gnmiconfig/worker.go]: That was for debugging, forgot to remove them.
Removed.

File: testkit/test_kit.go (hunk 2 above)
Comment (boleifu) [reply testkit/test_kit.go]: Done

**Pull request merged**